compiler/test/runnable/test18916.d

```diff
@@ -11,7 +11,7 @@ struct Line
 
 void foo(Line line1 = __LINE__, int line2 = __LINE__, int line3 = int(__LINE__))
 {
-    assert(line1 == 12);
+    assert(line1 == 21);
     assert(line2 == 21);
     assert(line3 == 12);
```

Comment:
This can also be modified to print 21, however, according to @WalterBright in this [comment](https://github.com/dlang/dmd/pull/8310#issuecomment-393802212) this sort of construction is used to retain the C++ behavior of __LINE__

Reply:
We also have bugs about `__FUNCTION__.ptr` not resolving correctly so I don't think that's a good argument.